src/wp-includes/blocks.php

```diff
@@ -1263,9 +1263,9 @@ function build_query_vars_from_query_block( $block, $page ) {
 			}
 		}
 		if ( isset( $block->context['query']['sticky'] ) && ! empty( $block->context['query']['sticky'] ) ) {
-			$sticky = get_option( 'sticky_posts' );
+			$sticky = (array) get_option( 'sticky_posts' );
```

Comment:
I think it's okay without the casting. Did you encounter any error or a possible path that would cause an error?

Reply:
I don't think this typecast is needed. Would prefer to keep this a pure backport.

```suggestion
			$sticky = get_option( 'sticky_posts' );
```

Reply:
@ntsekouras no I did not but since get_option can theoretically return anything (e.g. `false` if the db entry is missing, anything if a pre_option_ filter is doing something bad) and that value is then passed to array_merge, I thought adding a little safety was in order...